src/Tasks/ManifestUtil/SecurityUtil.cs

```diff
@@ -670,8 +671,24 @@ private static void SignPEFile(X509Certificate2 cert, Uri timestampUrl, string p
                         // everything was fine
                         break;
                     case 1:
-                        // error, report it
-                        throw new ApplicationException(String.Format(CultureInfo.InvariantCulture, resources.GetString("SecurityUtil.SigntoolFail"), path, signTool.StandardError.ReadToEnd()));
+                        {
+                            // error, retry with /t if timestamp url was given
+                            if (timestampUrl != null)
+                            {
+                                startInfo.Arguments = GetCommandLineParameters(cert.Thumbprint, timestampUrl, path, useSha256, false);
+                                signTool2 = Process.Start(startInfo);
+                                signTool2.WaitForExit();
+                                while (!signTool2.HasExited)
+                                {
+                                    System.Threading.Thread.Sleep(50);
+                                }
+                                // if error code is not 1, then retry succeeded
+                                if (signTool2.ExitCode != 1)
```

Comment:
Should an WarningException be thrown if signTool2.ExitCode is 2?

Reply:
Done.